crypto/pkcs12/p12_mutl.c

```diff
@@ -127,31 +221,50 @@ static int pkcs12_gen_mac(PKCS12 *p12, const char *pass, int passlen,
     }
     (void)ERR_pop_to_mark();
 
-    md_size = EVP_MD_get_size(md);
+    keylen = EVP_MD_get_size(md);
     md_nid = EVP_MD_get_type(md);
-    if (md_size < 0)
+    if (keylen < 0)
         goto err;
-    if ((md_nid == NID_id_GostR3411_94
-         || md_nid == NID_id_GostR3411_2012_256
-         || md_nid == NID_id_GostR3411_2012_512)
+
+    if (pbmac1_md_nid != NID_undef && pkcs12_key_gen == NULL /* Verification */) {
```

Comment:
```suggestion
    if (pbmac1_md_nid != NID_undef && pkcs12_key_gen == NULL /* Verification */) {
```
Generally, NULL does not really mean verification but just that the key gen function is not explicitly supplied.
Yet in case `pbmac1_md_nid != NID_undef` the comment is correct.
Why do you distinguish this case here? 

Please provide a clarifying comment for this special treatment using `pkcs12_gen_pbmac1_verify` for verification, while for generation with `PKCS12_set_pbmac1_with_pbkdf2()`, the function `pkcs12_key_gen` is explicitly provided, namely the slightly different function `pkcs12_pbmac1_pbkdf2_key_gen()`.
I am confused also because `PKCS12_set_mac()`, which also generates a MAC, may involve PBMAC1 while it sets `pkcs12_key_gen` to NULL.

Reply:
Adjusted, thanks